File: src/boj/boj_16926.java
package boj;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class boj_16926 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int r = Integer.parseInt(st.nextToken());

        int[][] arr = new int[n][m];
        for (int cr = 0; cr < n; cr++) {
            st = new StringTokenizer(br.readLine());
            for (int cc = 0; cc < m; cc++) {
                arr[cr][cc] = Integer.parseInt(st.nextToken());
            }
        }

        for (int d = 0; d < Math.min(n, m) / 2; d++) {
            int cnt = 2 * (n - 2 * d) + 2 * (m - 2 * d - 2);
            for (int i = 0; i < r % cnt; i++) {
                rotate(n, m, arr, d);
            }
        }

        for (int cr = 0; cr < n; cr++) {
            for (int cc = 0; cc < m; cc++) {
                bw.append(String.valueOf(arr[cr][cc])).append(' ');
            }
            bw.newLine();
        }
        bw.newLine();
        bw.flush();
        bw.close();
        br.close();
    }

    static void rotate(int n, int m, int[][] arr, int depth) {
        int temp = arr[depth][depth];
        int nr = depth, nc = depth;
        do {
            if (nr == depth && nc + 1 <= m - 1 - depth) {
                arr[nr][nc] = arr[nr][++nc];
            } else if (nc == m - 1 - depth && nr + 1 <= n - 1 - depth) {
                arr[nr][nc] = arr[++nr][nc];
            } else if (nr == n - 1 - depth && depth <= nc - 1) {
                arr[nr][nc] = arr[nr][--nc];
            } else {
                arr[nr][nc] = arr[--nr][nc];
            }
        } while (!(nr == depth + 1 && nc == depth));
        arr[nr][nc] = temp;
    }
}

File: src/boj/boj_16926.kt
package boj_16926

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*
import kotlin.math.min

var n = 0
var m = 0
var r = 0
lateinit var array: Array<IntArray>

fun main() {
    init()
    repeat(r) {
        rotate()
    }
    printAnswer()
}

fun init() = with(BufferedReader(InputStreamReader(System.`in`))) {
    with(StringTokenizer(readLine())) {
        n = nextToken().toInt()
        m = nextToken().toInt()
        r = nextToken().toInt()
    }
    array = Array(n) {
        with(StringTokenizer(readLine())) {
            IntArray(m) {
                nextToken().toInt()
            }
        }
    }
    close()
}

fun rotate() {
    repeat(min(n, m) / 2) { i ->
        var r = i
        var c = i
        val temp = array[i][i]
        do {
            array[r][c] = if (r == i && c + 1 < m - i) {
                array[r][++c]
            } else if (c == m - i - 1 && r + 1 < n - i) {
                array[++r][c]
            } else if (r == n - i - 1 && i < c) {
                array[r][--c]
            } else {
                array[--r][c]
            }
        } while (!(r == i + 1 && c == i))
        array[r][c] = temp
    }
}

fun printAnswer() = with(BufferedWriter(OutputStreamWriter(System.out))) {
    for (r in 0 until n) {
        for (c in 0 until m) {
            append("${array[r][c]} ")
        }
        newLine()
    }
    flush()
    close()
}